scripts/cmake/golang.cmake

```diff
@@ -55,6 +55,7 @@ function(build_go_archive OUTPUT_NAME MODULE_FILE)
                     CGO_LDFLAGS="${GOBUILD_CGO_LDFLAGS}"
                     GOOS=${GOBUILD_GOOS}
                     GOARCH=${GOBUILD_GOARCH}
+                    GOROOT=$ENV{GOROOT}
```

Comment:
Unrelated nitpick, we should capture the GOROOT when configuring, i think.

Reply:
I would recommend that instead of capturing `GOROOT` I would point this to something like `${CMAKE_BUILD_DIR}/goroot` so that golang build stuff also gets cleared when we delete the build directory.